src/lib.rs

```diff
@@ -86,16 +85,37 @@ pub fn parse(raw_languages: &str) -> Vec<String> {
     let stripped_languages = raw_languages.to_owned().replace(' ', "");
     let language_strings: Vec<&str> = stripped_languages.split(',').collect();
     let mut languages: Vec<Language> = language_strings.iter().map(|l| Language::new(l)).collect();
-
     languages.sort();
-
     languages
         .iter()
         .map(|l| l.name.to_owned())
         .filter(|l| !l.is_empty())
         .collect()
 }
 
+/// Similar to [`parse`](parse) but with quality `f32` appended to notice if it is a default value.
+/// is used by [`intersection_with_qualification`](intersection_with_qualification) and
+/// [`intersection_ordered_with_qualification`](intersection_ordered_with_qualification).
+///
+/// # Example
+///
+/// ```
+/// use accept_language::parse_with_qualification;
+///
+/// let user_languages = parse_with_qualification("en-US, en-GB;q=0.5");
+/// assert_eq!(user_languages,vec![(String::from("en-US"), 1.0), (String::from("en-GB"), 0.5)])
+/// ```
+pub fn parse_with_qualification(raw_languages: &str) -> Vec<(String, f32)> {
```

Comment:
I'm wondering now if `parse`, `intersection`, and `intersection_ordered` should just call `parse_with_qualification`, `intersection_with_qualification`, and `intersection_ordered_with_qualification` and map over the return type to change it from `Vec<(String, f32)>` to `Vec<String>`? The function bodies are almost exactly the same, minus the `map` closure.

Additionally, I think `quality` is the more standard term over `qualification`, which would also shorten the names a little bit 😉

Reply:
I see your [point](https://en.wikipedia.org/wiki/Don't_repeat_yourself). But that will also means a slower implementation for the user with a frugal request without quality. Another option would be to re-write it with macro's, so you get the fastest implementation for with and without quality, but I've no experience in writing them: possibly it will become less readable as code?

Reply:
I'm not too familiar with macros either, so let's leave it as-is for now 😄